kaitai/kaitaistream.cpp

```diff
@@ -390,9 +396,8 @@ std::string kaitai::kstream::ensure_fixed_contents(std::string expected) {
     std::string actual = read_bytes(expected.length());
 
     if (actual != expected) {
-        std::cout << "Fixed contents mismatch!\n";
-        std::cout << actual << "\n";
-        std::cout << expected << "\n";
+        // NOTE: I think printing it outright is not best idea, it could contain non-ascii charactes like backspace and beeps and whatnot. It would be better to print hexlified version, and also to redirect it to stderr.
+        throw std::runtime_error("ensure_fixed_contents: actual data does not match expected data");
```

Comment:
Typically, this is solved by defining a custom exception type and passing actual vs expected to it. It will lay quietly there, and only where end-user would request something like `caught_exception.what()`, we'll render it into a message with hex representations.

For now, though, I guess, throwing a `std::runtime_error` is still much better than that cout placeholder.